contributing/giving-constructive-criticism.rst

```diff
@@ -0,0 +1,154 @@
+  "stupid"). Assume everyone is intelligent and well-meaning.
+
+* Be explicit. Remember people don't always understand your intentions online.
+
+* Don't use hyperbole. ("always", "never", "endlessly", "nothing", "worst",
+  "horrible", "terrible").
+
+Instead of saying "This is a horrible idea" use something like
+"This can introduce issues in the future. Please make sure this is safe to use."
+
+Asking for changes
+------------------
+
+Rarely something is perfect from the start, while the code itself is good.
+It may not be optimal or conform the Symfony coding style.
+
+Again, understand the author already spend time on the issue and asking
+for (small) changes may be misinterpreted or seen as a personal attack.
+
+Be thankful for there work (so far), stay positive and really help them
+to make the contribution a great one. *Especially if they are a first
+time contributor.*
+
+Use words like "Please", "Thank you" and "Could you" instead of making demands;
+
+* "Thank you for your work so far. I left some suggestions for improvement
+  to make the code more readable."
+
+* "Your code contains some coding-style problems, can you fix these before
+  we merge? Thank you"
+
+* "Please use 4 spaces instead of tabs", "This needs be on the previous line";
+
+  During a pull request review you can usually leave more then one comment,
+  you don't have to use "Please" all the time. But it wouldn't hurt.
+
+It may not seem like much, but saying "Thank you" does make others feel
+more welcome.
+
+Making jokes (using humor)
+--------------------------
+
+In short: Don't be a troll; This violates the Code of Conduct and may
+even get you banned!
+
+OK, lets be honest. We all make jokes, some are funny and some are not.
+They don't put anyone of, right? Well, who's to tell. What's funny to one
+is offensive to another.
+
+In short. Keep it real. There is nothing wrong with using a meme or emoji
+from time to time, but don't over do it. And even then, keep it friendly.
+
+**Don't use sarcasm for a serious topic, that's not something that belongs
+to the Symfony community.** And don't marginalize someone's problems;
+"Well I guess that's not suppose to happen? 😆 ".
+
+Even if someone's explanation is "inviting to joke about", it's a real
+problem to them. Making jokes about this doesn't help with solving
+there problem and only makes them *feel stupid*. Instead try to discover
+what the problem is really about.
+
+If emoji, animated gifs, or humor aren't you, don't force them.
+
+Final words
+-----------
+
+Don't feel bad if you "failed" to follow these tips. As long as your
+intentions were good and you didn't really offended or insulted anyone;
+you can explain you misunderstood, you didn't meant to marginalize or
+simply failed.
+
+But don't say it "just because", if your apology is not really meant
+you *will* loose credibility and respect from other developers.
+
+*Don't do unto others what you don't want others to do unto you.*
```

Comment:
This is probably the negative of what is intended?

Reply:
Not sure 🤔 it got this from Google. it's more a common knowledge or "Golden rule" comment.

Reply:
I think what you're aiming to reference is _"Do unto others as you would have them do to you"_